app/controllers/certificates_controller.rb

```diff
@@ -0,0 +1,77 @@
+class CertificatesController < ApplicationController
+  before_action :set_certificate, only: [:show, :destroy, :update, :edit]
+  before_action :set_volunteer, except: [:edit, :show, :destroy]
+
+  def index
+    @certificates = Certificate.where(volunteer: @volunteer)
+    authorize Certificate
+  end
+
+  def show; end
+
+  def new
+    @certificate = Certificate.new(volunteer_id: @volunteer.id, user_id: current_user.id)
+    authorize @certificate
+  end
+
+  def create
+    @certificate = Certificate.new(certificate_params.merge(volunteer_id: @volunteer.id,
+      user_id: current_user.id))
+    @certificate.assignment_kinds = assignment_kinds
+    @certificate.text_body = certificate_params['text_body']
+    authorize @certificate
+    if @certificate.save
+      redirect_to volunteer_certificate_path(@volunteer, @certificate)
+    else
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    @certificate.update(certificate_params.merge(volunteer_id: @volunteer.id,
+      user_id: current_user.id))
+    @certificate.update(assignment_kinds: assignment_kinds,
+      text_body: certificate_params['text_body'],
+      volunteer_contact: certificate_params['volunteer_contact'],
+      institution: certificate_params['institution'])
+    authorize @certificate
+    if @certificate.save
+      redirect_to volunteer_certificate_path(@volunteer, @certificate)
+    else
+      render :new
+    end
+  end
+
+  def destroy
+    @certificate.destroy
+    redirect_to volunteer_path(@volunteer)
+  end
+
+  private
+
+  def assignment_kinds
+    certificate_params['assignment_kinds']
+      .to_h
+      .map { |key, bool| [key.to_sym, bool.to_i == 1] }
+      .to_h
+  end
+
+  def set_certificate
+    @certificate = Certificate.find(params[:id])
+    @volunteer = @certificate.volunteer
+    authorize @certificate
+  end
+
+  def set_volunteer
+    @volunteer = Volunteer.find(params[:volunteer_id])
+  end
+
+  def certificate_params
+    params.require(:certificate).permit(
+      :duration, :duration_end, :duration_start, :hours, :minutes, :text_body, :institution,
+      :funktion, volunteer_contact: [:name, :street, :city], assignment_kinds: Assignment::KINDS
```

Comment:
NIT picking, but since we use English for naming, shouldn't it be :function?

Reply:
If you want to bother to translate the whole certificate into entlish, go ahead. But as this Certificate is only available in German, French, and Italian there is somewhat zero sense to this.

Reply:
I was not talking about the document, of course. We have only German documents. 
I was talking about the attribute `:funktion`.

Reply:
As it is specifically for that form, I will not invent a new one. This is not nit picking. Sorry... It is pointless.
It is sort of impossible that this thing is ever used for a other language than german, at least not as is. So the effort to rename db fields, i simply refuse to take here.